src/phoenix/trace/openai/instrumentor.py

```diff
@@ -626,21 +640,42 @@ def _accumulate_messages(
                 content_token_lists[choice_index].append(content_token)
             if function_call := choice.delta.function_call:
                 if function_name := function_call.name:
-                    function_names[choice_index] = function_name
+                    function_calls[choice_index]["name"] = function_name
                 if (function_argument_token := function_call.arguments) is not None:
-                    function_argument_token_lists[choice_index].append(function_argument_token)
+                    function_calls[choice_index]["argument_tokens"].append(function_argument_token)
             if role := choice.delta.role:
                 roles[choice_index] = role
-    messages: List[OpenInferenceMessage] = [{} for _ in range(num_choices)]
+            if choice.delta.tool_calls:
+                for tool_call in choice.delta.tool_calls:
+                    tool_index = tool_call.index
+                    if not tool_call.function:
+                        continue
+                    if (name := tool_call.function.name) is not None:
+                        tool_calls[choice_index][tool_index]["name"] = name
+                    if (arguments := tool_call.function.arguments) is not None:
+                        tool_calls[choice_index][tool_index]["argument_tokens"].append(arguments)
+
+    messages: List[OpenInferenceMessage] = []
     for choice_index in range(num_choices):
+        message: Dict[str, Any] = {}
         if (role_ := roles.get(choice_index)) is not None:
-            messages[choice_index][MESSAGE_ROLE] = role_
-        if content_token_list := content_token_lists[choice_index]:
-            messages[choice_index][MESSAGE_CONTENT] = "".join(content_token_list)
-        if (function_name := function_names.get(choice_index)) is not None:
-            messages[choice_index][MESSAGE_FUNCTION_CALL_NAME] = function_name
-        if function_argument_token_list := function_argument_token_lists[choice_index]:
-            messages[choice_index][MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON] = "".join(
-                function_argument_token_list
-            )
+            message[MESSAGE_ROLE] = role_
+        if content_tokens := content_token_lists[choice_index]:
+            message[MESSAGE_CONTENT] = "".join(content_tokens)
+        if function_call_ := function_calls.get(choice_index):
+            if (name := function_call_.get("name")) is not None:
+                message[MESSAGE_FUNCTION_CALL_NAME] = name
+            if argument_tokens := function_call_.get("argument_tokens"):
+                message[MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON] = "".join(argument_tokens)
+        if tool_calls_ := tool_calls.get(choice_index):
+            message[MESSAGE_TOOL_CALLS] = [{} for _ in range(len(tool_calls_))]
+            for tool_index, tool_call_ in tool_calls_.items():
```

Comment:
i think you may want to use enumerate instead to ward off any potential index out of bound errors
```suggestion
            for tool_index, tool_call_ in enumerate(tool_calls_.values()):
```

Reply:
addressed in df18361a